src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -413,6 +387,51 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
         return new $class();
     }
 
+    /**
+     * @param array                $data
+     * @param string               $key
+     * @param \ReflectionParameter $constructorParameter
+     * @param array                $context
+     * @param string|null          $format
+     *
+     * @return mixed value of constructor parameter - an argument
+     *
+     * @throws LogicException
+     * @throws RuntimeException
+     * @throws MissingConstructorArgumentsException
+     */
+    protected function createConstructorArgument(array &$data, string $key, \ReflectionParameter $constructorParameter, array &$context, string $format = null)
+    {
+        $parameterData = $data[$key];
+        if (null === $parameterData && $constructorParameter->allowsNull()) {
+            // Don't run set for a parameter passed to the constructor
+            unset($data[$key]);
```

Comment:
The `&$data` argument makes little sense to me (especially given the name of the method). Can you move `unset($data[$key])` in the caller instead and directly pass `$parameterData`?

Reply:
Good point. Fixed. Maybe w can do similar with `$key` parameter? It's only used in exception message. Unless it's important to show normalized constructor parameter name, we can use `$constructorParameter->name`.